### PR TITLE
Remove configmap/secret validation and add PauseReconcilliation option

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/14821

### CODE REVIEW

File: pkg/app/app.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ type Config struct {
 	AppName             string
 	AppNamespace        string
 	AppVersion          string
-	ConfigVersion       string
+	ConfigMajorVersion  int
```
Comment (kopiczko) [inline pkg/app/app.go:21]: Why that change? That will require changes in the architect binary as well. I'd rather drop this eventually. We probably aren't gonna use that.

Comment (kubasobon) [reply pkg/app/app.go:21]: PTAL at revert commit: https://github.com/giantswarm/app/pull/62/commits/af2302e693c271cd91e0074407ddf50f1c2d9c34

Should I change it back to ConfigVersion string and keep the rest of the revert intact?

Comment (kopiczko) [reply pkg/app/app.go:21]: I don't remember how it was included in a single PR, but it should be `ConfigVersion string`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- Set `config.giantswarm.io/version` using expected value format.
```
Comment (kopiczko) [inline CHANGELOG.md:12]: Changelog and PR title should need update.

**Comment by kubasobon** [comment]
PTAL

File: pkg/app/app.go [hunk 3]
```diff
@@ -37,6 +38,9 @@ func NewCR(c Config) *applicationv1alpha1.App {
 		if !c.DisableForceUpgrade {
 			annotations["chart-operator.giantswarm.io/force-helm-upgrade"] = "true"
 		}
+		if c.PauseReconciliation {
+			annotations["app-operator.giantswarm.io/paused"] = "true"
```
Comment (rossf7) [inline pkg/app/app.go:42]: Can we use the constant Pawel is adding? https://github.com/giantswarm/apiextensions/pull/649

**Review by rossf7 — APPROVED** [review]
Nice, LGTM

Just would like us to use the const from apiextensions for the annotation.

File: pkg/app/app.go (hunk 3 above)
Comment (kubasobon) [reply pkg/app/app.go:42]: Sure thing.

File: CHANGELOG.md [hunk 4]
```diff
@@ -24,8 +32,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Change (unused yet) `config.giantswarm.io/major-version` annotation to `config.giantswarm.io/version`.
 
-## [3.6.0] - 2020-12-01
```
Comment (kopiczko) [inline CHANGELOG.md:27]: Did something go wrong here?

Comment (kubasobon) [reply CHANGELOG.md:27]: A bad revert perhaps. I'll bring it back.

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add `PauseReconciliation` option, responsible for setting `app-operator.giantswarm.io/paused` flag.
+
+### Removed
+
+- Validate App CR configmap and secret names if managed by config-controller.
```
Comment (kopiczko) [inline CHANGELOG.md:16]:
```suggestion
- Do not validate App CR configmap and secret names if managed by config-controller.
```

**Review by kopiczko — APPROVED** [review]

**Pull request merged**